File: 003/Problem003.py
''' 
  * Problem003.py
  *
  * Project Euler: Problem 3
  * 
  * Problem: Find the largest prime factor of the number 600851475143
  *          
  *
'''

import math

def largest_prime_factor(n):
    largest = 1
    
    # if even, remove factors of 2
    if n % 2 == 0:
        largest, n = 2, n/2
    
    # odd factors
    p, current = n, 3
    while n != 1 and current <= math.sqrt(p):
        while n % current == 0:
            largest = current
            n /= current
        current += 2
    
    return largest

print largest_prime_factor(600851475143)

File: 003/Problem003.java
public class Problem003 {
    public static void main( String[] args ) {
        int result = largestPrimeFactor(600851475143L);
        System.out.println(result);

    }

    private static int largestPrimeFactor(long n) {
        int largest = 1;
        int temp = 0;
        if (n % 2 == 0) {
            largest = 2;
            n /= 2;
        }
        long p = n;
        int current = 3;
        while (n != 1 && current <= Math.sqrt(p)) {
            while (n % current == 0) {
                largest = current;
                n /= current;
            }
            current += 2;
        }
        return largest;
    }
}
